### PR TITLE
interp: support implicit-type slice of pointers of struct composite literal

### LINKED ISSUE #883: interp: typeless initialization of slice of pointers of struct does not work
The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
	"image"
)

func main() {
	v := []*image.Point{
		{X: 3, Y: 2},
	}
	fmt.Println(v)
}
```

Expected result:
```console
$ go run ./sample.go
// [(3,2)]
```

Got:
```console
$ yaegi ./sample.go
cfg.go:2395: _test/struct_slice0.go:10:3: compositeGenerator not implemented for type kind: ptr
_test/struct_slice0.go:10:3: CFG post-order panic: _test/struct_slice0.go:10:3: compositeGenerator not implemented for type kind: ptr
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002f4600, 0x40)
	/Users/mpl/go1/src/runtime/debug/stack.go:24 +0x9d
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc000219d08)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:451 +0xbc
panic(0x1a36820, 0xc00009efa0)
	/Users/mpl/go1/src/runtime/panic.go:967 +0x15d
github.com/traefik/yaegi/interp.(*Interpreter).cfg.func2.1(0xc000477500)
	/Users/mpl/src/github.com/traefik/yaegi/interp/cfg.go:424 +0xb3
panic(0x194fca0, 0xc00009c7f0)
	/Users/mpl/go1/src/runtime/panic.go:967 +0x15d
log.Panic(0xc000218f28, 0x1, 0x1)
	/Users/mpl/go1/src/log/log.go:351 +0xac
github.com/traefik/yaegi/interp.compositeGenerator(0xc000477500, 0xc000377b20, 0x1)
	/Users/mpl/src/github.com/traefik/yaegi/interp/cfg.go:2395 +0x1eb
github.com/traefik/yaegi/interp.compositeGenerator(0xc000477500, 0xc000377a40, 0x0)
	/Users/mpl/src/github.com/traefik/yaegi/interp/cfg.go:2366 +0x54
github.com/traefik/yaegi/interp.(*Interpreter).cfg.func2(0xc000477500)
	/Users/mpl/src/github.com/traefik/yaegi/interp/cfg.go:985 +0xaba5
github.com/traefik/yaegi/interp.(*node).Walk(0xc000477500, 0xc000219b88, 0xc000219b58)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:218 +0xaf
github.com/traefik/yaegi/interp.(*node).Walk(0xc000476f00, 0xc000219b88, 0xc000219b58)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:215 +0x70
github.com/traefik/yaegi/interp.(*node).Walk(0xc000476d00, 0xc000219b88, 0xc000219b58)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:215 +0x70
github.com/traefik/yaegi/interp.(*node).Walk(0xc000476c00, 0xc000219b88, 0xc000219b58)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:215 +0x70
github.com/traefik/yaegi/interp.(*node).Walk(0xc000476700, 0xc000219b88, 0xc000219b58)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:215 +0x70
github.com/traefik/yaegi/interp.(*node).Walk(0xc000476000, 0xc000219b88, 0xc000219b58)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:215 +0x70
github.com/traefik/yaegi/interp.(*Interpreter).cfg(0xc000372000, 0xc000476000, 0xc000357b00, 0x4, 0xc000357b00, 0x4, 0x0, 0x0, 0xc000476000)
	/Users/mpl/src/github.com/traefik/yaegi/interp/cfg.go:63 +0x20b
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc000372000, 0xc0002d6bd0, 0x8b, 0x7ffeefbff6c2, 0x16, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:478 +0x2ed
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc000372000, 0x7ffeefbff6c2, 0x16, 0x200, 0xc0002d6b40, 0x8b, 0x0, 0xc0000aef70)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:392 +0x102
main.runFile(0xc000372000, 0x7ffeefbff6c2, 0x16, 0xc000096360, 0x1)
	/Users/mpl/src/github.com/traefik/yaegi/cmd/yaegi/run.go:113 +0x158
main.run(0xc0000321d0, 0x1, 0x1, 0x1acb1b0, 0x1968060)
	/Users/mpl/src/github.com/traefik/yaegi/cmd/yaegi/run.go:77 +0x6e2
main.main()
	/Users/mpl/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:119 +0x2f4

run: _test/struct_slice0.go:10:3: CFG post-order panic: _test/struct_slice0.go:10:3: compositeGenerator not implemented for type kind: ptr
```

N.B: this works fine if the type is specified, i.e. with `&image.Point{X: 3, Y: 2},`

### PR DESCRIPTION
**Pull request by mpl**
Also fix same case for implicit-type maps of etc.

Fixes #883

### CODE REVIEW

File: interp/cfg.go [hunk 1]
```diff
@@ -975,14 +975,15 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
 					vtyp := &itype{cat: valueT, rtype: rtype.Elem()}
 					err = check.mapLitExpr(child, ktyp, vtyp)
 				}
+				// TODO(mpl): how come we don't have anything to do for reflect.Ptr here?
```
Comment (mvertes) [inline interp/cfg.go:978]: Nothing more to do here, as a composite expression always applies directly to the target type. A pointer indirection, if any,  is handled in the parent AST node. At runtime (run.go) of course the pointer indirection must be handled.

Comment (mvertes) [reply interp/cfg.go:978]: you can safely remove this comment

Comment (mvertes) [reply interp/cfg.go:978]: you can safely remove this comment

**Review by mvertes — CHANGES_REQUESTED** [review]

File: interp/cfg.go (hunk 1 above)
Comment (mpl) [reply interp/cfg.go]: done

**Review by mvertes — APPROVED** [review]

**Pull request merged**